contracts/iden3/verifiers/BaseVerifier.sol

```diff
@@ -0,0 +1,111 @@
+    using EnumerableSet for EnumerableSet.UintSet;
+    using SetHelper for EnumerableSet.UintSet;
+
+    IZKPQueriesStorage public zkpQueriesStorage;
+
+    // schema => allowed issuers
+    mapping(uint256 => EnumerableSet.UintSet) internal _allowedIssuers;
+
+    function __BaseVerifier_init(IZKPQueriesStorage zkpQueriesStorage_) internal onlyInitializing {
+        __Ownable_init();
+
+        _setZKPQueriesStorage(zkpQueriesStorage_);
+    }
+
+    function setZKPQueriesStorage(
+        IZKPQueriesStorage newZKPQueriesStorage_
+    ) external override onlyOwner {
+        _setZKPQueriesStorage(newZKPQueriesStorage_);
+    }
+
+    function updateAllowedIssuers(
+        uint256 schema_,
+        uint256[] calldata issuerIds_,
+        bool isAdding_
+    ) external override onlyOwner {
+        _updateAllowedIssuers(schema_, issuerIds_, isAdding_);
+    }
+
+    function getAllowedIssuers(uint256 schema_) public view override returns (uint256[] memory) {
+        return _allowedIssuers[schema_].values();
+    }
+
+    function isAllowedIssuer(
+        uint256 schema_,
+        uint256 issuerId_
+    ) public view virtual override returns (bool) {
+        return _allowedIssuers[schema_].contains(issuerId_);
+    }
+
+    function _setZKPQueriesStorage(IZKPQueriesStorage newZKPQueriesStorage_) internal {
+        zkpQueriesStorage = newZKPQueriesStorage_;
+    }
+
+    function _updateAllowedIssuers(
+        uint256 schema_,
+        uint256[] calldata issuerIds_,
+        bool isAdding_
+    ) internal {
+        if (isAdding_) {
+            _allowedIssuers[schema_].add(issuerIds_);
+        } else {
+            _allowedIssuers[schema_].remove(issuerIds_);
+        }
+    }
+
+    function _transitState(TransitStateParams memory transitStateParams_) internal {
+        ILightweightState lightweightState_ = zkpQueriesStorage.lightweightState();
+        ILightweightState.GistRootData memory currentGISTRootData_ = lightweightState_
+            .getCurrentGISTRootInfo();
+
+        if (
+            !lightweightState_.isIdentitiesStatesRootExists(
+                transitStateParams_.newIdentitiesStatesRoot
+            ) &&
+            currentGISTRootData_.createdAtTimestamp <
+            transitStateParams_.gistData.createdAtTimestamp
+        ) {
+            lightweightState_.signedTransitState(
+                transitStateParams_.newIdentitiesStatesRoot,
+                transitStateParams_.gistData,
+                transitStateParams_.proof
+            );
+        }
+    }
+
+    function _checkAllowedIssuer(string memory queryId_, uint256 issuerId_) internal view virtual {
+        require(
+            isAllowedIssuer(zkpQueriesStorage.getStoredSchema(queryId_), issuerId_),
+            "BaseVerifier: Issuer is not on the list of allowed issuers."
+        );
+    }
+
+    function _checkChallenge(uint256 challenge_) internal view virtual {
+        require(
```

Comment:
msg.sender is the voting contract here

Reply:
Deleted the _checkChallenge check in favor of the onlyVoting modifier